From 4eccf890ee6a84fad85bc917fbc5dca1b3b697bd Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Fri, 7 Mar 2025 17:22:45 +0100
Subject: [PATCH] force x86_64 host arch detection

---
 makepanda/makepandacore.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index 74a4307a29..151285c49e 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -337,7 +337,7 @@ def GetHostArch():
     """Returns the architecture we're compiling on.
     Its value is also platform-dependent, as different platforms
     have different architecture naming."""
-
+    return 'x86_64' # when cross-compiling, conda python is faking the target environment, so platform.machine() would returns an incorrect host arch
     target = GetTarget()
     if target == 'windows':
         return 'x64' if host_64 else 'x86'
-- 
2.34.1

